src/main/java/com/openshift/internal/restclient/model/DeploymentConfig.java

```diff
@@ -164,4 +188,79 @@ public IDeploymentTrigger addTrigger(String type) {
 	public String getDeploymentStrategyType() {
 		return asString(DEPLOYMENTCONFIG_STRATEGY);
 	}
+
+	@Override
+	public boolean haveTriggersFired() {
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes == null || causes.getType() != ModelType.LIST)
+			return false;
+		return causes.asList().size() > 0;
+	}
+
+	@Override
+	public boolean didImageTrigger(String imageTag) {
+		if (!haveTriggersFired())
+			return false;
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		for (ModelNode cause : causes.asList()) {
+			String type = cause.get(TYPE).asString();
+			if (type.equalsIgnoreCase(IMAGE_CHANGE)) {
+				String triggerName = cause.get(IMAGE_TRIGGER).get(FROM).get(NAME).asString();
+				if (triggerName.contains("/")) {
+					StringTokenizer st = new StringTokenizer(imageTag, "/");
+					String tmp = null;
+					while (st.hasMoreTokens()) {
+						tmp = st.nextToken();
+					}
+					if (imageTag.equals(tmp)) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public String getImageHexIDForImageTag(String imageTag) {
+		ModelNode triggers = get(DEPLOYMENTCONFIG_TRIGGERS);
+		if (triggers == null || triggers.getType() != ModelType.LIST)
+			return null;
+		for (ModelNode trigger : triggers.asList()) {
+			if (trigger.get(TYPE).asString().equalsIgnoreCase(IMAGE_CHANGE)) {
+				String tag = null;
+				tag = trigger.get(IMAGE_CHANGE_PARAMS).get(FROM).get(NAME).asString();
+				if (imageTag.equals(tag)) {
+					return trigger.get(IMAGE_CHANGE_PARAMS).get(LAST_TRIGGER_IMAGE).asString();
+				}
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public String getImageTagForTriggeredDeployment() {
+		String imageTag = null;
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes == null || causes.getType() != ModelType.LIST)
+			return null;
+		for (ModelNode cause : causes.asList()) {
+			String type = cause.get(TYPE).asString();
+			if (type.equalsIgnoreCase(IMAGE_CHANGE)) {
+				imageTag = cause.get(IMAGE_TRIGGER).get(FROM).get(NAME).asString();
+				if (imageTag.contains("/")) {
+					StringTokenizer st = new StringTokenizer(imageTag, "/");
```

Comment:
switched to DockerImageURI (which uses split()) per comment from @jcantrill - StringTokenizer gone.